### ISSUE TITLE
Changes to docker apt repository broke non-ubuntu instances

### ISSUE THREAD

**Issue by snowe2010** [LemmyNet/lemmy-ansible#123]
We're completely unable to install using lemmy-ansible due to the changes in commit d2403a1976c7c21241cbd39ebad66b70df369ca6 which force usage of Ubuntu.

**Comment by eslerm** [comment]
Before https://github.com/LemmyNet/lemmy-ansible/commit/d2403a1976c7c21241cbd39ebad66b70df369ca6 all installs used the 20.04 binary of docker.

Which distro do you need support for?

**Comment by ismailkarsli** [comment]
> Before [d2403a1](https://github.com/LemmyNet/lemmy-ansible/commit/d2403a1976c7c21241cbd39ebad66b70df369ca6) all installs used the 20.04 binary of docker.
> 
> Which distro do you need support for?

I'm using Debian 11 and getting the error above:
```TASK [Download Docker GPG Key] *******************************************************************************************
fatal: [root@128.140.48.219]: FAILED! => {"changed": false, "checksum_dest": null, "checksum_src": "f5b5bd1487cefc0c53c947e11ca202e86b33dbad", "dest": "/etc/apt/keyrings/docker.asc", "elapsed": 0, "msg": "Destination /etc/apt/keyrings does not exist", "src": "/root/.ansible/tmp/ansible-moduletmp-1691581922.179723-24vaqrm8/tmp8hlvs__1", "url": "https://download.docker.com/linux/debian/gpg"}```

**Comment by snowe2010** [comment]
sorry, I somehow missed the notification. I'm also using Debian 11. I had to completely revert the commit that changed that. It wasn't due to the docker image being ubuntu, it was like @ismailkarsli said, the when stopped this step from ever being run.

**Comment by snowe2010** [comment]
why have the check for Ubuntu (or as you're suggesting, Debian) at all? Everything would continue to work fine without that check. Until logic is added for other non-apt systems it's just breaking all the other apt systems from working with lemmy-ansible.

**Comment by codyro** [comment]
The `lemmy-ansible` project should have a few target distributions that it supports to make maintaining and supporting easier to avoid running into issues like this. 

I suggest starting by offering the following targets and modifying the layout to be a bit more modular (EX; https://github.com/LemmyNet/lemmy-ansible/pull/145#issuecomment-1671639359, or similar):

* Ubuntu 22.04 LTS
* AlmaLinux/RHEL/CentOS/RockyLinux 9
* Debian 10/11

I can assist in this.

**Comment by eslerm** [comment]
Debian 11 might still be using `/etc/apt/keyrings` instead of `/etc/apt/trusted.gpg.d`. The former should not be used in newer Ubuntu releases.

Before the checks, https://github.com/LemmyNet/lemmy-ansible/pull/116, every lemmy-ansible installed the docker binary compiled for Ubuntu 20.04 on every distro.

**Comment by codyro** [comment]
> Debian 11 might still be using `/etc/apt/keyrings` instead of `/etc/apt/trusted.gpg.d`. The former should not be used in newer Ubuntu releases.

I just took a peak. Should be safe to move to `/etc/apt/trusted.gpg.d` if we're only looking to target the aforementioned distributions.

```bash
root@aa17e97a8841:/# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@aa17e97a8841:/# ls -lhd /etc/apt/trusted.gpg.d/
drwxr-xr-x. 2 root root 4.0K Jul 25 00:00 /etc/apt/trusted.gpg.d/
```

```bash
root@94186324bc8e:/# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@94186324bc8e:/# ls -lhd /etc/apt/trusted.gpg.d
drwxr-xr-x. 2 root root 4.0K Jul 25 00:00 /etc/apt/trusted.gpg.d```

**Comment by therealmaxmoon** [comment]
I can confirm that it doesn't work for Debian 11 (bullseye). I get the same error @ismailkarsli mentioned:

> ```
> fatal: [root@128.140.48.219]: FAILED! => {"changed": false, "checksum_dest": null, "checksum_src": "f5b5bd1487cefc0c53c947e11ca202e86b33dbad", "dest": "/etc/apt/keyrings/docker.asc", "elapsed": 0, "msg": "Destination /etc/apt/keyrings does not exist", "src": "/root/.ansible/tmp/ansible-moduletmp-1691581922.179723-24vaqrm8/tmp8hlvs__1", "url": "https://download.docker.com/linux/debian/gpg"}
> ```

I am still on Debian 11, because I still had my old server running and wanted to test Lemmy again, but I might reinstall it with Debian 12 (bookworm) if I decide to run a productive instance of Lemmy.

Will the installation work on bookworm with the current lemmy.yml?

**Comment by codyro** [comment]
@therealmaxmoon Please try using the changes in https://github.com/LemmyNet/lemmy-ansible/pull/148 :)

**Comment by therealmaxmoon** [comment]
> @therealmaxmoon Please try using the changes in #148 :)

I've downloaded [your lemmy.yml](https://raw.githubusercontent.com/codyro/lemmy-ansible/deb-arch-issue130/lemmy.yml) and the problem above doesn't appear anymore in Debian 11.

But I am facing another error, even if it's only restarting nginx (but this could be a problem of my old server):

```
TASK [reload nginx with new config] ********************************************************************************************************************************************************************************
fatal: [xxx]: FAILED! => {"changed": true, "cmd": "nginx -s reload", "delta": "0:00:00.005013", "end": "2023-08-09 19:31:10.105818", "msg": "non-zero return code", "rc": 127, "start": "2023-08-09 19:31:10.100805", "stderr": "/bin/sh: 1: nginx: not found", "stderr_lines": ["/bin/sh: 1: nginx: not found"], "stdout": "", "stdout_lines": []}
```

**Comment by codyro** [comment]
That should likely be reported in another issue to be fixed. Looks like `nginx` isn't in the default `PATH`--are you running this as an unprivileged user, perchance? I believe, by default, it's located in `/usr/sbin/nginx`, which might explain why it's not finding it.

That line is just passing a signal to `nginx` to reload the configuration: `nginx -s reload`

**Comment by snowe2010** [comment]
Your fix worked for me.

**Comment by codyro** [comment]
This issue should now be safe to close, as a fix was merged in #148
